metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -121,6 +127,182 @@ final class BloopServers(
+    val source = scala.io.Source.fromFile(
+      Paths
+        .get(System.getProperty("user.home"))
+        .resolve(".bloop/created_by_metals.lock")
+        .toFile
+    )
+    val value: Long = source.mkString.toLong
+    source.close()
+    value
+  }.getOrElse(0)
+
+  private def processUserPreferenceForBloopJvmProperties(
+      messageActionItem: MessageActionItem,
+      bloopGlobalJsonFile: File,
+      requestedBloopJvmProperties: List[String],
+      javaHome: String,
+      reconnect: () => Future[BuildChange]
+  ): Future[Unit] = {
+    messageActionItem match {
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.applyAndRestart =>
+        writeJVMPropertiesToBloopGlobalJsonFile(
+          bloopGlobalJsonFile,
+          requestedBloopJvmProperties,
+          javaHome
+        ) match {
+          case Failure(exception) => Future.failed(exception)
+          case Success(_) =>
+            shutdownServer()
+            reconnect().ignoreValue
+        }
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.saveButNotRestart =>
+        writeJVMPropertiesToBloopGlobalJsonFile(
+          bloopGlobalJsonFile,
+          requestedBloopJvmProperties,
+          javaHome
+        ) match {
+          case Failure(exception) => Future.failed(exception)
+          case Success(_) => Future.successful(())
+        }
+
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.openGlobalJsonFile =>
+        val showDocumentParams = new ShowDocumentParams()
+        showDocumentParams.setUri(bloopGlobalJsonFile.toURI.toString)
+        languageClient.showDocument(showDocumentParams).asScala.ignoreValue
+      case item
+          if item == Messages.BloopGlobalJsonFilePremodified.useGlobalFile =>
+        Future.successful(())
+    }
+  }
+
+  private def updateBloopGlobalJsonFileThenRestart(
+      bloopGlobalJsonFile: File,
+      requestedBloopJvmProperties: List[String],
+      javaHome: String,
+      reconnect: () => Future[BuildChange]
+  ): Future[Unit] = {
+    writeJVMPropertiesToBloopGlobalJsonFile(
+      bloopGlobalJsonFile,
+      requestedBloopJvmProperties,
+      javaHome
+    ) match {
+      case Failure(exception) => Future.failed(exception)
+      case Success(_) =>
+        languageClient
+          .showMessageRequest(
+            Messages.BloopJvmPropertiesChange.params()
+          )
+          .asScala
+          .flatMap {
+            case messageActionItem
+                if messageActionItem == Messages.BloopJvmPropertiesChange.reconnect =>
+              shutdownServer()
+              reconnect().ignoreValue
+            case _ =>
+              Future.successful(())
+          }
+    }
+
+  }
+
+  private def getBloopGlobalJsonFile(): Option[File] =
+    Try {
+      Paths
+        .get(System.getProperty("user.home"))
+        .resolve(".bloop/bloop.json")
+        .toFile
+    }.toOption
```

Comment:
```suggestion
  private def getBloopGlobalJsonFile(): Option[AbsolutePath] =
     sys.props.get("user.home").map{
       home =>
         AbsolutePath(Paths.get(home).resolve(".bloop/bloop.json"))
     }
```
it will be much easier to use since it already contains a bunch of utility methods.